bundles/org.openhab.binding.volvooncall/src/main/java/org/openhab/binding/volvooncall/internal/handler/VehicleHandler.java

```diff
@@ -173,16 +173,26 @@ private void queryApiAndUpdateChannels() {
             } catch (JsonSyntaxException | IOException e) {
                 logger.warn("Exception occurred during execution: {}", e.getMessage(), e);
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
-                ScheduledFuture<?> refreshJob = this.refreshJob;
-                if (refreshJob != null) {
-                    refreshJob.cancel(true);
-                    refreshJob = null;
-                }
+                freeRefreshJob();
                 startAutomaticRefresh(configuration.refresh);
             }
         }
     }
 
+    private void freeRefreshJob() {
+        ScheduledFuture<?> refreshJob = this.refreshJob;
+        if (refreshJob != null) {
+            refreshJob.cancel(true);
+            this.refreshJob = null;
+        }
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        freeRefreshJob();
```

Comment:
As a general rule, super.dispose() should go last.
```suggestion
        freeRefreshJob();
        super.dispose();
```